src/mpyl/steps/steps.py

```diff
@@ -112,9 +112,16 @@ def _execute(
 
     @staticmethod
     def _find_required_artifact(
-        project: Project, stages: list[Stage], required_artifact: Optional[ArtifactType]
+        project: Project,
+        stages: list[Stage],
+        required_artifact: Optional[ArtifactType],
+        deploy_image: Optional[str] = None,
     ) -> Optional[Artifact]:
-        if not required_artifact or required_artifact == ArtifactType.NONE:
+        if (
+            not required_artifact
+            or required_artifact == ArtifactType.NONE
+            or (required_artifact == ArtifactType.DOCKER_IMAGE and deploy_image)
```

Comment:
Am I reading this wrong, or should this be 
```suggestion
            or (required_artifact == ArtifactType.DOCKER_IMAGE and not deploy_image)
```
?

Reply:
Ah no, ignore the previous comment.

Reply:
([This method is planned to go away soon anwyay](https://github.com/Vandebron/gh-mpyl/pull/77) 😄)

Reply:
What a rollercoaster of comments! Haha